packages/plugin/src/theia-proposed.d.ts

```diff
@@ -734,4 +739,57 @@ export module '@theia/plugin' {
     }
 
     // #endregion
+
+    // #region Status bar item with ID and Name: https://github.com/microsoft/vscode/issues/74972
+
+    /**
+     * Options to configure the status bar item.
+     */
+    export interface StatusBarItemOptions {
+
+        /**
+         * A unique identifier of the status bar item. The identifier
+         * is for example used to allow a user to show or hide the
+         * status bar item in the UI.
+         */
+        id: string;
+
+        /**
+         * A human readable name of the status bar item. The name is
+         * for example used as a label in the UI to show or hide the
+         * status bar item.
+         */
+        name: string;
+
+        /**
+         * Accessibility information used when screen reader interacts with this status bar item.
+         */
+        accessibilityInformation?: AccessibilityInformation;
+
+        /**
+         * The alignment of the status bar item.
+         */
+        alignment?: StatusBarAlignment;
+
+        /**
+         * The priority of the status bar item. Higher value means the item should
+         * be shown more to the left.
+         */
+        priority?: number;
+    }
+
+    export namespace window {
+
+        /**
+         * Creates a status bar [item](#StatusBarItem).
+         *
+         * @param options The options of the item. If not provided, some default values
+         * will be assumed. For example, the `StatusBarItemOptions.id` will be the id
+         * of the extension and the `StatusBarItemOptions.name` will be the extension name.
+         * @return A new status bar item.
+         */
+        export function createStatusBarItem(options?: StatusBarItemOptions): StatusBarItem;
```

Comment:
@sgraband similar to above, where does this api come from?

Reply:
Similar to above. See [here](https://github.com/microsoft/vscode/blob/3c4e3df9e89829dce27b7b5c24508306b151f30d/src/vs/vscode.proposed.d.ts#L877-L888).